wallet/reservation.c

```diff
@@ -654,6 +654,94 @@ static const struct json_command fundpsbt_command = {
 };
 AUTODATA(json_command, &fundpsbt_command);
 
+static struct command_result *json_newoutput(struct command *cmd,
+					    const char *buffer,
+					    const jsmntok_t *obj UNNEEDED,
+					    const jsmntok_t *params)
+{
+	struct json_stream *response;
+	struct amount_sat *amount;
+	struct wally_psbt *psbt;
+	u32 *locktime;
+	ssize_t outnum;
+	u32 weight;
+	struct pubkey pubkey;
+	s64 keyidx;
+	u8 *b32script;
+
+	if (!param(cmd, buffer, params,
+		   p_req("satoshi", param_sat, &amount),
+		   p_opt("initialpsbt", param_psbt, &psbt),
+		   p_opt("locktime", param_number, &locktime),
+		   NULL))
+		return command_param_failed();
+
+	if (!psbt) {
+		if (!locktime) {
+			locktime = tal(cmd, u32);
+			*locktime = default_locktime(cmd->ld->topology);
+		}
+		psbt = create_psbt(cmd, 0, 0, *locktime);
```

Comment:
nit: use `create_psbt(cmd, 0, 1, *locktime);` to pre-allocate the space for the output. Doesn't really matter for the single output you add here but its a good habit to get into when multiple inputs/outputs are possible.

Reply:
In this case the code has to handle making its own PSBT and modifying the one the user passes in. Since we have to have the code for dynamically adding an output anyway, I think it's better to stick with that code for both paths than have an extra static path.